wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneResolver.swift

```diff
@@ -92,68 +87,95 @@ public final class OneOnOneResolver: OneOnOneResolverInterface {
     ) async throws -> OneOnOneConversationResolution {
         WireLogger.conversation.debug("resolving 1-1 conversation with user: \(userID)")
 
-        let messageProtocol = try await protocolSelector.getProtocolForUser(
-            with: userID,
-            in: context
-        )
+        let messageProtocol = try await protocolSelector.getProtocolForUser(with: userID, in: context)
 
         switch messageProtocol {
         case .none:
-            WireLogger.conversation.debug("no common protocols found")
-            await context.perform {
-                guard
-                    let otherUser = ZMUser.fetch(with: userID, in: context),
-                    let conversation = otherUser.oneOnOneConversation
-                else {
-                    return
-                }
+            return await resolveCommonUserProtocolNone(with: userID, in: context)
+        case .mls:
+            return try await resolveCommonUserProtocolMLS(with: userID, in: context)
+        case .proteus:
+            return resolveCommonUserProtocolProteus()
+        case .mixed:
```

Comment:
This comes from the fact that we are using the same type for two unrelated things. We should probably split it into something like  `ConversationProtocol` and `UserSupportedProtocol`.

Not now of course.